### ISSUE TITLE
Does the Sagemaker toolkit predict_fn() contain with torch.no_grad() by default?

### ISSUE THREAD

**Issue by pleonova** [huggingface/notebooks#275]
This is more of a question, rather an issue, but I was wondering @philschmid if you happen to know if the default implementation of the Sagmaker toolkit has `with torch.no_grad()` already?
https://github.com/huggingface/notebooks/blob/d0b1a33565fc595fa9423b2c247f36abf6f0ede1/sagemaker/17_custom_inference_script/code/inference.py#L28

When I was working the Google CoLab environment, and using HF pipelines, the models didn't seem to zero out the gradients on inference and I ran into issues. However, when I implemented it in pytorch and added the `with torch.no_grad()', I saw huge speedups. I didn't see those same speedups while using the Sagemaker toolkit, both off the shelf and the custom one you showed in your notebook, so was curious if that is already accounted for?
